agent/src/capability/servicemanifestfileloader.cpp

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (C) 2016-2017 Pelagicore AB
+ *
+ * Permission to use, copy, modify, and/or distribute this software for
+ * any purpose with or without fee is hereby granted, provided that the
+ * above copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+ * WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+ * BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+ * OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+ * WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+ * ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+ * SOFTWARE.
+ *
+ * For further information see LICENSE
+ */
+
+#include "servicemanifestfileloader.h"
+
+namespace softwarecontainer {
+
+ServiceManifestFileLoader::ServiceManifestFileLoader(const std::string &source) :
+    ServiceManifestLoader(source)
+{
+}
+
+std::vector<json_t *> ServiceManifestFileLoader::loadContent()
+{
+    std::string errorMessage;
+
+    if (m_source.empty()) {
+        //File path is empty, this is ok
+        log_debug() << "Path to service manifests is empty";
+    } else if (m_source.compare("/") == 0) {
+        errorMessage = "Searching for configuration files from root dir not allowed";
+        log_error() << errorMessage;
+        throw ServiceManifestPathError(errorMessage);
+    } else if (isDirectory(m_source)) {
+        log_debug() << "Path to service manifests is a dir";
+        loadServiceManifestDir();
+    } else if (isFile(m_source)) {
+        log_debug() << "Path to service manifest is a file";
+        loadServiceManifestFile(m_source);
+    } else {
+        errorMessage = "Path to service manifest does not exist: \"" + m_source + "\"";
```

Comment:
Actually, this is not true. It could be a socket or a fifo (just not a directory or file). I would rewrite this error message.